pages/src/calc.rs

```diff
@@ -0,0 +1,111 @@
+use crate::compression::{Compression, CompressionSet};
+use crate::mode::Mode;
+use crate::row::Row;
+use std::collections::HashSet;
+use std::fmt::{Display, Formatter};
+
+#[derive(Copy, Clone)]
+pub enum Bottleneck {
+    Bandwidth,
+    Cpu,
+}
+
+impl Display for Bottleneck {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Bandwidth => "bandwidth",
+                Self::Cpu => "CPU",
+            }
+        )
+    }
+}
+
+pub struct CalcRow {
+    pub compression: Compression,
+    pub crate_: String,
+    pub limit: Bottleneck,
+    pub messages_per_second: f32,
+    pub relative: f32,
+}
+
+pub fn calc(
+    rows: Vec<Row>,
+    messages_per_benchmark: u32,
+    bandwidth: u64,
+    cpus: f32,
+    compression_set: &CompressionSet,
+    mode: Mode,
+) -> Vec<CalcRow> {
+    let bandwidth_per_second = bandwidth as f32 * (1.0 / (30.4 * 24.0 * 60.0 * 60.0));
+
+    let mut rows: Vec<_> = rows
+        .into_iter()
+        .flat_map(|r| {
+            let Row {
+                crate_,
+                serialize,
+                deserialize,
+                sizes,
+                ..
+            } = r;
+            let deserialize = deserialize.unwrap_or(0.0);
```

Comment:
Won't this skew results in favor of libs that don't support deserialization when the comparison mode is set to serialize + deserialize round trip and most libs are CPU-bound? Setting the bandwidth to 3Tb/Mo on the `log` bench under `round trip` shows this particular failure mode.